<!DOCTYPE html>
<html>
<head>
<title>README</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<style type="text/css">
/* GitHub stylesheet for MarkdownPad (http://markdownpad.com) */
/* Author: Nicolas Hery - http://nicolashery.com */
/* Version: b13fe65ca28d2e568c6ed5d7f06581183df8f2ff */
/* Source: https://github.com/nicolahery/markdownpad-github */

/* RESET
=============================================================================*/

html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, menu, nav, output, ruby, section, summary, time, mark, audio, video {
  margin: 0;
  padding: 0;
  border: 0;
}

/* BODY
=============================================================================*/

body {
  font-family: Helvetica, arial, freesans, clean, sans-serif;
  font-size: 14px;
  line-height: 1.6;
  color: #333;
  background-color: #fff;
  padding: 20px;
  max-width: 960px;
  margin: 0 auto;
}

body>*:first-child {
  margin-top: 0 !important;
}

body>*:last-child {
  margin-bottom: 0 !important;
}

/* BLOCKS
=============================================================================*/

p, blockquote, ul, ol, dl, table, pre {
  margin: 15px 0;
}

/* HEADERS
=============================================================================*/

h1, h2, h3, h4, h5, h6 {
  margin: 20px 0 10px;
  padding: 0;
  font-weight: bold;
  -webkit-font-smoothing: antialiased;
}

h1 tt, h1 code, h2 tt, h2 code, h3 tt, h3 code, h4 tt, h4 code, h5 tt, h5 code, h6 tt, h6 code {
  font-size: inherit;
}

h1 {
  font-size: 28px;
  color: #000;
}

h2 {
  font-size: 24px;
  border-bottom: 1px solid #ccc;
  color: #000;
}

h3 {
  font-size: 18px;
}

h4 {
  font-size: 16px;
}

h5 {
  font-size: 14px;
}

h6 {
  color: #777;
  font-size: 14px;
}

body>h2:first-child, body>h1:first-child, body>h1:first-child+h2, body>h3:first-child, body>h4:first-child, body>h5:first-child, body>h6:first-child {
  margin-top: 0;
  padding-top: 0;
}

a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
  margin-top: 0;
  padding-top: 0;
}

h1+p, h2+p, h3+p, h4+p, h5+p, h6+p {
  margin-top: 10px;
}

/* LINKS
=============================================================================*/

a {
  color: #4183C4;
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

/* LISTS
=============================================================================*/

ul, ol {
  padding-left: 30px;
}

ul li > :first-child, 
ol li > :first-child, 
ul li ul:first-of-type, 
ol li ol:first-of-type, 
ul li ol:first-of-type, 
ol li ul:first-of-type {
  margin-top: 0px;
}

ul ul, ul ol, ol ol, ol ul {
  margin-bottom: 0;
}

dl {
  padding: 0;
}

dl dt {
  font-size: 14px;
  font-weight: bold;
  font-style: italic;
  padding: 0;
  margin: 15px 0 5px;
}

dl dt:first-child {
  padding: 0;
}

dl dt>:first-child {
  margin-top: 0px;
}

dl dt>:last-child {
  margin-bottom: 0px;
}

dl dd {
  margin: 0 0 15px;
  padding: 0 15px;
}

dl dd>:first-child {
  margin-top: 0px;
}

dl dd>:last-child {
  margin-bottom: 0px;
}

/* CODE
=============================================================================*/

pre, code, tt {
  font-size: 12px;
  font-family: Consolas, "Liberation Mono", Courier, monospace;
}

code, tt {
  margin: 0 0px;
  padding: 0px 0px;
  white-space: nowrap;
  border: 1px solid #eaeaea;
  background-color: #f8f8f8;
  border-radius: 3px;
}

pre>code {
  margin: 0;
  padding: 0;
  white-space: pre;
  border: none;
  background: transparent;
}

pre {
  background-color: #f8f8f8;
  border: 1px solid #ccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px;
}

pre code, pre tt {
  background-color: transparent;
  border: none;
}

kbd {
    -moz-border-bottom-colors: none;
    -moz-border-left-colors: none;
    -moz-border-right-colors: none;
    -moz-border-top-colors: none;
    background-color: #DDDDDD;
    background-image: linear-gradient(#F1F1F1, #DDDDDD);
    background-repeat: repeat-x;
    border-color: #DDDDDD #CCCCCC #CCCCCC #DDDDDD;
    border-image: none;
    border-radius: 2px 2px 2px 2px;
    border-style: solid;
    border-width: 1px;
    font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
    line-height: 10px;
    padding: 1px 4px;
}

/* QUOTES
=============================================================================*/

blockquote {
  border-left: 4px solid #DDD;
  padding: 0 15px;
  color: #777;
}

blockquote>:first-child {
  margin-top: 0px;
}

blockquote>:last-child {
  margin-bottom: 0px;
}

/* HORIZONTAL RULES
=============================================================================*/

hr {
  clear: both;
  margin: 15px 0;
  height: 0px;
  overflow: hidden;
  border: none;
  background: transparent;
  border-bottom: 4px solid #ddd;
  padding: 0;
}

/* TABLES
=============================================================================*/

table th {
  font-weight: bold;
}

table th, table td {
  border: 1px solid #ccc;
  padding: 6px 13px;
}

table tr {
  border-top: 1px solid #ccc;
  background-color: #fff;
}

table tr:nth-child(2n) {
  background-color: #f8f8f8;
}

/* IMAGES
=============================================================================*/

img {
  max-width: 100%
}
</style>
</head>
<body>
<h1>README</h1>
<h1>Introduction</h1>
<p>The <code>kcUID</code> software suite is used to process UID library RNA-seq reads in SeqHealth. <code>kcUID</code> suite have been compiled with C++ and are executable in the directory. If any questions, please contract SeqHealth.</p>
<h1>Depends</h1>
<p>The suite includes <code>kcUID</code>, <code>IdentifyUIDs</code>, <code>ClusterUID</code>, <code>CallConsensus</code>, <code>CorrectUID</code>, <code>Line2Fast</code>, <code>FastCount</code> and <code>UIDSummary</code>. </p>
<p>It will also invoke some Linux commands, such as <code>pandoc</code>, <code>awk</code>, <code>wc</code>, <code>sort</code> and <code>rm</code> and please make sure these commands are in your PATH. </p>
<p>The suite also relies on <code>R</code> to generate figs and html report. The <code>kcUIDReport_0.1.1.tar.gz</code> is the R package for <code>kcuid</code>. Before analysis, the <code>kcUIDReport</code> should be first installed in R which depends on <code>ggplot2</code>, <code>reshape2</code>, <code>rmarkdown</code>, <code>htmltools</code>, <code>readr</code>, <code>DT</code>, <code>knitr</code> and <code>kableExtra</code> packages.</p>
<h1>Usage</h1>
<p>The <strong>kcUID</strong> executable program is <strong>the main entry point</strong>. It will invoke the follow-up analyses and control the whole process. Try <code>kcUID --help</code> to see more helps.</p>
<p>The steps which <code>kcUID</code> will invoke are shown below.</p>
<ol>
<li>
<p>Identify UID in reads: This step invokes <code>IdentifyUIDs</code> to identify UIDs from every read1.</p>
<p>Commands as: </p>
<pre><code>IdentifyUIDs -r1 test.clean.R1.fastq -r2 test.clean.R2.fastq -o test.uid.txt  -ur1 test.not.R1.fastq -ur2 test.not.R2.fastq -n 5
</code></pre>

<p>The output 'test.uid.txt' will include UID, ReadID, Read1 with UID removed and Read2 if r2 is provided.</p>
</li>
<li>
<p>Sort UID: This step will sort the UIDs in the uid output.</p>
<p>Command as: </p>
<pre><code>sort --parallel 4 --buffer-size G -k 1,1 -o test.sort.uid.txt test.uid.txt
</code></pre>

</li>
<li>
<p>Cluster UID in every UID. A uid may contain some different molecules, so the step is to cluster similar sequences to form clusters within the same UID. </p>
<p>Command as</p>
<pre><code>ClusterUID -i test.sort.uid.txt -o test.clu.txt -t 10 -n 5'
</code></pre>

<p>The output will include UID, sub-cluster ID in the same UID, Read1 and Read2 if r2 is provided.</p>
</li>
<li>
<p><code>UIDSummary</code> is used to summay UID usage and cluster size distribution.</p>
</li>
<li>
<p>Get consensus sequence in every UID sub-cluster. For every sub-cluster, we will conduct multiple sequence alignment (MSA) and get an consensus sequence.</p>
<p>Command as: </p>
<pre><code>CallConsensus -i test.clu.txt -o test.con.txt -n 5
</code></pre>

</li>
<li>
<p>During PCR or sequencing, errors may happen in UIDs. So after consensus generation, we merged all of the identical consensus to form unique consensus sequences among similar uids.</p>
<p>Command as: </p>
<pre><code>sort --parallel 4 --buffer-size G -k 3,4 -o test.sort.con.txt test.con.txt
CorrectUID -i test.sort.con.txt -o test.cor.con.txt
</code></pre>

</li>
<li>
<p>The output in the previous step is one line including read1 consensus and read2 consensus. But we always need fasta or fastq for downstream analysis. So change txt to fasta/fastq.</p>
<p>Command as: </p>
<pre><code>Line2Fast -i test.cor.con.txt -p test.con -type fasta
or  
Line2Fast -i test.cor.con.txt -p test.con -type fastq
</code></pre>

</li>
<li>
<p><code>FastCount</code> is used to get reads information such as length distribution, base distribution, duplication level. It reads fasta or fastq files and algorithms are similar with FastQC.</p>
</li>
</ol>
<h1>Result</h1>
<p>After running <code>kcUID</code>, you will get such results:</p>
<ol>
<li>
<p>*.dedup.R1.fastq/fasta: The reads1 that have been dedup.</p>
</li>
<li>
<p>*.dedup.R2.fastq/fasta: The reads2 that have been dedup (if read2 has been provided).</p>
</li>
<li>
<p>*.noUID.R1.fastq/fasta：The reads1 that don't contain UIDs.</p>
</li>
<li>
<p>*.noUID.R2.fastq/fasta：The reads2 that don't contain UIDs.</p>
</li>
<li>
<p>*.kcUID.report.html: The kcUID html report.</p>
</li>
</ol>
<p><center>Copyright ©2019 SeqHealth, WuHan, China</center></p>

</body>
</html>
<!-- This document was created with MarkdownPad, the Markdown editor for Windows (http://markdownpad.com) -->
